modules/exploits/windows/http/exchange_proxyshell_rce.rb

```diff
@@ -480,6 +476,42 @@ def execute_command(cmd, _opts = {})
   end
 end
 
+# Use https://learn.microsoft.com/en-us/exchange/client-developer/web-service-reference/resolvenames to resolve mailbox
+# information. The endpoint only returns 100 at a time though so if the target has more than that many email addresses
+# multiple requests will need to be made. Since the endpoint doesn't support pagination, we refine the query by using
+# progressively larger search prefixes until there are less than 101 results and thus will fit into a single response.
+class MailboxEnumerator
+  def initialize(mod)
+    @mod = mod
+  end
+
+  # the characters that Exchange Server 2019 allows in an alias (no unicode)
+  ALIAS_CHARSET = 'abcdefghijklmnopqrstuvwxyz0123456789!#$%&\'*+-/=?^_`{|}~'.freeze
```

Comment:
This was pulled straight from ECP, it is not case sensitive.

![image](https://user-images.githubusercontent.com/2058303/205395441-157a3794-3c8e-4233-a8df-83e9d4c0b8a4.png)

> Strings formed with characters from A to Z (uppercase or lowercase), digits from 0 to 9, !, #, $, %, &, ', *, +, -, /, =, ?, ^, _, `, {, |, } or ~. One or more periods may be embedded in an alias, but each period should be preceded and followed by at least one of the other characters. Unicode characters from U+00A1 to U+00FF are also valid in an alias, but they will be mapped to a best-fit US-ASCII string in the email address, which is generated from such an alias.